packages/design-system/src/components/text-field.tsx

```diff
@@ -147,7 +147,11 @@ const TextFieldBase = styled("div", {
 });
 
 type TextFieldProps = Omit<
-  React.ComponentProps<typeof TextFieldBase> & React.ComponentProps<"input">,
+  Omit<
```

Comment:
In general, this can be definitely improved - it would be good if you'd have a common solution for this kind of type merge. Note though that `Omit` also doesn't come for free - especially for such huge types as the ones here (those DOM-oriented types have over 200 properties).